File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/PassportDataFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram;

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface PassportDataFilter {

    AnnotationState status() default AnnotationState.ON;

    String[] validator() default {};

    EncryptedPassportElementArrayFilter encryptedPassportElements() default @EncryptedPassportElementArrayFilter(status = AnnotationState.OFF);

    EncryptedCredentialsFilter encryptedCredentials() default @EncryptedCredentialsFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/PassportDataFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class PassportDataFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val encryptedPassportElements: EncryptedPassportElementArrayFilter = EncryptedPassportElementArrayFilter(status = AnnotationState.OFF), val encryptedCredentials: EncryptedCredentialsFilter = EncryptedCredentialsFilter(status = AnnotationState.OFF))